src/mscorlib/shared/System/SpanDebugView.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal sealed class SpanDebugView<T>
+    {
+        private readonly T[] _array;
+
+        public SpanDebugView(Span<T> span)
+        {
+           _array = span.ToArray();
```

Comment:
I am not sure if making copy will work. You should talk to the debugger people or try on some other type wrapping an array. The reason is that I think the debugger instantiates just one proxy and then calls the properties on it multiple times (when it needs to refresh the view).

Reply:
I don't understand. If the instantiation only happens once and only the properties item gets called, then wouldn't this issue cause problems while debugging **slow span** too (see [SpanDebugView](https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanDebugView.cs#L17))? 

Can you provide a code sample that would cause issues?

```C#
byte[] array = new byte[5] { 1, 2, 3, 4, 5 };

string[] strArray = new string[6] { "12", "ab", "agfs", "AGadg", "ag", "" };

var span = new Span<byte>(array, 1, 3);
Console.WriteLine(span.Length);

span = span.Slice(2);

var rospan = new ReadOnlySpan<byte>(array, 1, 3);
Console.WriteLine(rospan.Length);

var span2 = new Span<string>(strArray, 1, 3);
Console.WriteLine(span2.Length);
}
```
![image](https://user-images.githubusercontent.com/6527137/32570337-b1935558-c478-11e7-9256-b4a1e3e9b88b.png)


![image](https://user-images.githubusercontent.com/6527137/32570366-c4963666-c478-11e7-8811-eb6ba9be1ca9.png)